Mage/src/main/java/mage/target/common/TargetAnyTargetAmount.java

```diff
@@ -14,25 +14,20 @@ public class TargetAnyTargetAmount extends TargetPermanentOrPlayerAmount {
     private static final FilterPermanentOrPlayer defaultFilter
             = new FilterAnyTarget("targets");
 
-    public TargetAnyTargetAmount(int amount) {
-        this(amount, 0);
+    public TargetAnyTargetAmount(int amount, int minNumberOfTargets) {
```

Comment:
I'd consider making this constructor only take amount as a param, and set minNumberOfTargets as `(amount > 3 ? 0 : 1)` which appears to be accurate for all 38 usages currently. Any that don't align can use the three argument constructor for clarity (currently that's just Sundering Stroke)

If you agree, also change the TargetPermanentAmount subclasses similarly, and add doc comments for all to clarify usage

Reply:
I did initially consider that but had decided against it, since I feared there was a high probability of creating subtle bugs in the future. The bad scenario I imagined was that someone implementing a new card would copy code from a similar card with a different `amount`, and miss the difference in the target requirements.

But, thinking about it again, if someone did make that error, I'd like to think that someone would notice that the generated card text was wrong (assuming the text didn't need to be set manually). So maybe I was being overly paranoid? I'll happily add amount-only constructors for common uses if you think it'd be okay.

Reply:
Good points. I've gone ahead and put back in simpler constructors where relevant, and wrote some documentation for the card-facing `TargetAmount` subclasses.